src/main/java/org/elasticsearch/rest/BytesRestResponse.java

```diff
@@ -120,9 +120,19 @@ public RestStatus status() {
     }
 
     private static XContentBuilder convert(RestChannel channel, RestStatus status, Throwable t) throws IOException {
-        XContentBuilder builder = channel.newBuilder().startObject()
-                .field("error", detailedMessage(t))
-                .field("status", status.getStatus());
+        XContentBuilder builder = channel.newBuilder().startObject();
+        if (channel.emitStackTrace()) {
+            buildDetailedErrorField(channel, t, builder);
+        } else {
+            buildSimpleErrorField(t, builder);
+        }
+        builder.field("status", status.getStatus());
+        builder.endObject();
+        return builder;
+    }
+
+    private static void buildDetailedErrorField(RestChannel channel, Throwable t, XContentBuilder builder) throws IOException {
+        builder.field("error", detailedMessage(t));
         if (t != null && channel.request().paramAsBoolean("error_trace", false)) {
```

Comment:
I actually meant an error (http 400)? Messages can be ignored.  This is just an invalid configuration.